src/ngraph/runtime/cpu/cpu_builder.cpp

```diff
@@ -360,6 +365,63 @@ namespace ngraph
             }
 
+            template <>
+            void Builder::BUILDER_DECL(ngraph::runtime::cpu::op::ConvertLayout)
+            {
+                auto& functors = external_function->get_functors();
+
+                auto& arg_tensor = external_function->get_tensor_data(args[0].get_name());
+                auto& out_tensor = external_function->get_tensor_data(out[0].get_name());
+
+                auto& mkldnn_emitter = external_function->get_mkldnn_emitter();
+
+                auto input_desc = mkldnn_utils::get_input_mkldnn_md(node, 0);
+                auto result_desc = mkldnn_utils::get_output_mkldnn_md(node, 0);
+
+                if (input_desc.data.format == mkldnn_nchw &&
+                    result_desc.data.format == mkldnn_goihw)
+                {
+                    //becomes a copy
+                    input_desc = result_desc;
+                }
+                else if (input_desc.data.format == mkldnn_nchw && input_desc.data.ndims == 4 &&
+                         result_desc.data.ndims == 5 && node->get_users().size() == 1)
+                {
+                    Shape weights_shape_groups;
+                    if (auto gconv = std::dynamic_pointer_cast<ngraph::op::GroupConvolution>(
+                            node->get_users()[0]))
+                    {
+                        weights_shape_groups = gconv->get_weights_dimensions();
+                    }
+                    else if (auto gconvb =
+                                 std::dynamic_pointer_cast<ngraph::op::GroupConvolutionBias>(
+                                     node->get_users()[0]))
+                    {
+                        weights_shape_groups = gconvb->get_weights_dimensions();
+                    }
+                    else
+                    {
+                        throw ngraph_error("Incompatible input/output shape in ConvertLayout op");
+                    }
+                    input_desc = mkldnn::memory::desc(
+                        mkldnn::memory::dims(weights_shape_groups.begin(),
+                                             weights_shape_groups.end()),
+                        mkldnn_utils::get_mkldnn_data_type(args[0].get_element_type()),
+                        mkldnn::memory::format::goihw);
+                }
+
+                size_t reorder_index = mkldnn_emitter->build_reorder(input_desc, result_desc);
+
+                auto& deps = mkldnn_emitter->get_primitive_deps(reorder_index);
+                auto functor = [&, reorder_index](CPURuntimeContext* ctx,
+                                                  CPUExecutionContext* ectx) {
+                    cpu::mkldnn_utils::set_memory_ptr(ctx, deps[0], arg_tensor);
+                    cpu::mkldnn_utils::set_memory_ptr(ctx, deps[1], out_tensor);
+                    cpu::mkldnn_utils::mkldnn_invoke_primitive(ctx, reorder_index);
+                };
+                functors.emplace_back(functor);
+            }
+
```

Comment:
This seems like a hack. Can we root-cause and do the right thing?

Reply:
The reason of the redefined issue is because when cpu_builder.cpp using the following lines to register the build_dispatcher .  {TI(ngraph::runtime::cpu::op::ConvertLayout),
-                     &runtime::cpu::Builder::build<ngraph::runtime::cpu::op::ConvertLayout>}, MSVC create builder function for ConvertLayout using template function in cpu_build.hpp line 265. later it conflict with the function in builder/convert_layout.cpp. Is there other way to avoid this if not moving the explicit function definition before the use?

Reply:
We can remove this line. This was probably accidentally left over from when we didn't have the registration mechanism.

Reply:
I pushed a better fix by creating a new template called REGISTER_CPU_OP_BUILDER and register the build function in convert_layout.cpp. The fix follows the design of keeping complex builder functions in separate file.  @jbobba Please take a look.